public/components/alerts/alerts.js

```diff
@@ -1,57 +1,94 @@
+import Kefir from 'kefir';
 import Component from 'can-component';
 import DefineMap from 'can-define/map/';
-import template from './alerts.stache';
+import 'can-define-stream';
+import view from './alerts.stache';
 import './alerts.less';
 import hub from '../../lib/hub';
 import AlertItem from '../../models/alert';
 
-/**
- * TODO: use streams for modifying alerts
- *      - https://github.com/donejs/bitcentive/issues/149
- */
+export const reducers = {
+  add(alerts, alert) {
+    return [alert, ...alerts.slice()];
+  },
+  remove(alerts, alert) {
+    return alerts.filter(item => item.id !== alert.id);
+  }
+};
 
 export const ViewModel = DefineMap.extend({
-  subscription: '*',
-  alerts: {
-    Value: AlertItem.List
+  addAlertStream: {
+    value() {
+      return this.stream('add-alert')
+        .map(ev => ev.args[0])
+        .merge(Kefir.stream(emitter => {
+          const subscription = hub.subscribe('alert', alert => emitter.emit(alert));
+          return () => {
+            subscription.remove();
+          };
+        }))
+        .map(alert => {
+          if ( !(alert instanceof AlertItem) ) {
+            alert = new AlertItem(alert);
+          }
+          return { 
+            action: 'ADD_ALERT', 
+            alert
+          };
+        });
+    }
   },
-  addAlert (alert) {
-    this.alerts.push(alert);
+  autoHideStream: {
+    value() {
+      return this.addAlertStream.flatMap(ev => {
+        if (ev.alert.displayInterval > 0) {
+          return Kefir.later(ev.alert.displayInterval, {
+            action: 'REMOVE_ALERT',
+            alert: ev.alert
+          });
+        } 
+        return Kefir.constant({ action: 'no-op' });
+      });
+    }
   },
-  removeAlert (alert) {
-    const idx = this.alerts.indexOf(alert);
-    if (idx !== -1) {
-      this.alerts.splice(idx, 1);
+  removeAlertStream: {
+    value() {
+      return this.stream('remove-alert').map(ev => {
+        return {
+          action: 'REMOVE_ALERT',
+          alert: ev.args[0]
+        };
+      });
     }
   },
-  // called by ($inserted) event
-  showAlert (alert) {
-    // timeout allows for paint cycle to pass
-    setTimeout(() => {
-      alert.visible = true;
-    }, 30);
-    
-    if (alert.displayInterval > 0) {
-      setTimeout(() => this.hideAlert(alert), 30 + alert.displayInterval);
+  alerts: {
+    stream() {
+      // TODO: use can-event
+      return Kefir.merge([this.addAlertStream, this.autoHideStream, this.removeAlertStream]).scan((alerts, ev) => {
```

Comment:
`this.stream("alert").merge( this. autoHideStream ).merge( this.stream("remove") )`